### DISCUSSION TITLE
升级UMI3的实时编译性能问题

### DISCUSSION THREAD

**Discussion by phoema** [umijs/umi#6387] [Q&A]
UMI3.4.8，保存一下js文件，实时编译需要30秒左右
UMI2.13.17，保存一下js文件，实时编译需要1秒左右
以下是编辑记录，请问差距这么大的原因是什么，实时编译时间可以回到UMI2吗？

UMI3.4.8
Compiled successfully in 30.92s
 DONE  Compiled successfully in 30947ms                                                                                                                                                               13:53:04
 WAIT  Compiling...                                                                                                                                                                                   13:53:09
√ Webpack
  Compiled successfully in 38.38s
 DONE  Compiled successfully in 38381ms                                                                                                                                                               13:53:48
 WAIT  Compiling...                                                                                                                                                                                   14:23:45
√ Webpack
  Compiled successfully in 26.20s
 DONE  Compiled successfully in 26202ms                                                                                                                                                               14:24:12

UMI2.13.17

 WAIT  Compiling...
Build completed in 1.763s
 DONE  Compiled successfully in 1764ms
 WAIT  Compiling...
Build completed in 0.493s
 DONE  Compiled successfully in 493ms
 WAIT  Compiling...
Build completed in 1.052s
 DONE  Compiled successfully in 1053ms
 WAIT  Compiling...
 WAIT  Compiling...                                                                                                                                                                              14:44:16
Build completed in 0.486s
 DONE  Compiled successfully in 487ms                                                                                                                                                            14:44:17
 WAIT  Compiling...                                                                                                                                                                              14:47:32
Build completed in 1.514s
 DONE  Compiled successfully in 1515ms                                                                                                                                                           14:47:33

**Comment by xiaohuoni** [comment]
应该是存在异常，你可以删除缓存文件重试下。如果依旧存在，可以提供可复现的用例，利于定位问题。
